src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/integration/IntegrationTestBase.kt

```diff
@@ -101,16 +109,34 @@ abstract class IntegrationTestBase {
     roles: List<String> = listOf(),
   ): (HttpHeaders) -> Unit = jwtAuthHelper.setAuthorisation(user, roles)
 
+  @BeforeEach
+  fun `Clear queues`() {
+    domainEventsQueue.sqsClient.purgeQueue(PurgeQueueRequest.builder().queueUrl(domainEventsQueue.queueUrl).build())
+    reset(telemetryClient)
+  }
+
   fun getNumberOfMessagesCurrentlyOnQueue(): Int? = domainEventsQueue.sqsClient.countMessagesOnQueue(domainEventsQueueUrl).get()
 
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
     private val localStackContainer = LocalStackContainer.instance
+    private val postgresContainer = PostgresContainer.instance
 
     @JvmStatic
     @DynamicPropertySource
-    fun testcontainers(registry: DynamicPropertyRegistry) {
+    fun containers(registry: DynamicPropertyRegistry) {
       localStackContainer?.also { setLocalStackProperties(it, registry) }
+
+      val url = postgresContainer?.let { postgresContainer.jdbcUrl } ?: DB_DEFAULT_URL
+      log.info("Using TestContainers?: ${postgresContainer != null}, DB url: $url")
+      registry.add("spring.datasource.url") { url }
+      registry.add("spring.datasource.username") { DB_USERNAME }
+      registry.add("spring.datasource.password") { DB_PASSWORD }
+      registry.add("spring.flyway.url") { url }
+      registry.add("spring.flyway.user") { DB_USERNAME }
+      registry.add("spring.flyway.password") { DB_PASSWORD }
+      registry.add("spring.datasource.placeholders.database_update_password") { DB_PASSWORD }
+      registry.add("spring.datasource.placeholders.database_read_only_password") { DB_USERNAME }
```

Comment:
mmh.. would it make sense to be consistent with localstack and have a `setPostgresProperties` on `PostgresContainer`?